src/proxy/inbound.rs

```diff
@@ -259,138 +291,173 @@ impl Inbound {
         metrics: Arc<Metrics>,
         socket_factory: Arc<dyn SocketFactory + Send + Sync>,
     ) -> Result<Response<Empty<Bytes>>, hyper::Error> {
-        match req.method() {
-            &Method::CONNECT => {
```

Comment:
the diff might look odd here because I un-indented this to be an `if { return } ...` rather than a match with a very long clause